src/cudadev/plugin-SiPixelClusterizer/SiPixelRawToClusterGPUKernel.cu

```diff
@@ -334,7 +329,7 @@ namespace pixelgpudetails {
   }
 
   // Kernel to perform Raw to Digi conversion
-  __global__ void RawToDigi_kernel(const SiPixelROCsStatusAndMapping *cablingMap,
+  __global__ void RawToDigi_kernel(const SiPixelROCsStatusAndMapping cablingMap,
```

Comment:
What would happen if the `const` would be missing along
```suggestion
  __global__ void RawToDigi_kernel(SiPixelROCsStatusAndMapping cablingMap,
```
? Would the function still get only read access to the memory, or full read-write access?

Reply:
The function would get read-write access to the memory. I actually see an improvement here where we can add `__restrict__` references  in the const use case and get the data cached through the non-coherent cache. I will look into this.

Reply:
I'm more worried that it is too easy (e.g. by mistake) to get a read-write access into something that should be treated as `const`. This is crucial for all consumers of Event or EventSetup data products, the constness of the data must be preserved (which is why there is `DeviceConstView` for the consumers e.g. here 
https://github.com/cms-patatrack/pixeltrack-standalone/blob/bb3fff74dbf8b34c6bd8ea7f8336e71faebce1fa/src/cudadev/CUDADataFormats/SiPixelDigisCUDA.h#L50
)

Reply:
I see. Const correctness should achieve this, but I will look into having an explicitly const version of the SoA structures.

Reply:
This feature is implemented in branch https://github.com/ericcano/pixeltrack-standalone/tree/macroBasedSoA-PR211-folowup. I will introduce it in a new PR and close this one.